Start & Stop Navigation in existing Map (#54)
maplibre/maplibre-navigation-ios: README.md

- Transitioned from the [Mapbox SDK](https://github.com/mapbox/mapbox-gl-native-ios) (version 4.3) to [Maplibre Maps SDK](https://github.com/maplibre/maplibre-gl-native) (version 6.0.0)
- Added optional config parameter in NavigationMapView constructor to customize certain properties like route line color

# Migrating from v2 to v3

> Maplibre v3 allows you to start a navigation in an existing Map, so no modal ViewController needs to be presented over an existing map as before. This results in some breaking changes. 

Reviewer: ```suggestion
MaplibreNavigation v3 allows you to start a navigation in an existing Map, so no modal ViewController needs to be presented over an existing map as before. This results in some breaking changes. 
```

### Step 1:

> Replace your ViewController that hosts the mapView with a `NavigationViewController`. We suggest to create a subclass of `NavigationViewController` and override init & call `super.init(dayStyle:)` or `super.init(dayStyleURL:)`. NavigationViewController will not do anything on its own until you start a navigation.

Reviewer: ```suggestion
Replace your ViewController that hosts the mapView with a `NavigationViewController`. We suggest to create a subclass of `NavigationViewController` and override init & call `super.init(dayStyle:)` or `super.init(dayStyleURL:)`. With v3, NavigationViewController will not start navigation until you request it.
```

### Step 2:

Start the navigation by calling `startNavigation(with: route)`. If you want to simulate your route, you need to pass in the optional locationManager parameter, otherwise the real location of the device will be used.

```swift
func locationManager(for route: Route) -> NavigationLocationManager {
#if targetEnvironment(simulator)
	let locationManager = SimulatedLocationManager(route: route)
	locationManager.speedMultiplier = 2
	return locationManager
#else

> 	return NavigationLocationManager()

Reviewer: ```
#if targetEnvironment(simulator) 
    let locationManager = SimulatedLocationManager(route: route)
    locationManager.speedMultiplier = 2
    self.startNavigation(with: route, locationManager: locationManager)
#else
    self.startNavigation(with: route)
#endif
```

Slightly shorter and we can avoid the concept of `NavigationLocationManager()` - people can just use the default value.

#endif
}

self.startNavigation(with: route, locationManager: locationManager(for : route))
```

### Step 3:

> Make your ViewController conform to `NavigationViewControllerDelegate` and implement `navigationViewControllerDidFinishRouting(_ navigationViewController: NavigationViewController)` to transition back to normal map mode. This delegate is called when the user arrives at the destination or cancels the navigation.

Reviewer 2: It's probably worth noting (not sure if here or elsewhere?) that the _similar_ method this replaces (dismissal; maybe via cancellation) has also changed semantics slightly. Internal logic used to depend on whether the method was implemented or not. This is no longer the case. I like the new version better, but think we should probably document this subtlety.

Author: This function `@objc optional func navigationViewControllerDidDismiss(_ navigationViewController: NavigationViewController, byCanceling canceled: Bool)` was removed. Will mention it in the README.

```swift
extension SceneDelegate: NavigationViewControllerDelegate {
    func navigationViewControllerDidFinishRouting(_ navigationViewController: NavigationViewController) {

>         navigationViewController.endNavigation()

Reviewer: `endNavigation` is a new method right?

Is there a reason `navigationViewController` doesn't call `endNavigation` itself, rather than requiring implementors to have a specific implementation for their delegate?

Author: I thought this gives the most flexibility. Two examples:
* when you want to display an arrived view.
* If you want to support multi stop routing, you could then start the next navigation without animating back and forth

Reviewer 2: It looks like it does call `endNavigation()` at least when I'm reading in right now.

    }
}
```

### Backwards compatibility

> If for some reason, you want to keep the old way of presenting a navigation modally, you can still do that. Simply call `startNavigation(with: route)` right after creating the `NavigationViewController`.

Reviewer 2: ```suggestion
If you want to keep the old way of presenting a navigation modally, you can still do that. Simply call `startNavigation(with: route)` right after creating the `NavigationViewController`.
```

- **Have a bug to report?** [Open an issue](https://github.com/maplibre/maplibre-navigation-ios/issues). If possible, include the version of Maplibre Services, a full log, and a project that shows the issue.
- **Have a feature request?** [Open an issue](https://github.com/maplibre/maplibre-navigation-ios/issues/new). Tell us what the feature should do and why you want the feature.

## Sample code

> We do provide a limited example app but its not functional right out of the box. 

Reviewer: 🎉

Reviewer 2: This is great!

Maybe you could elaborate slightly on what exactly it does / does not do in the README.

ISC is meant to be functionally equivalent to the MIT license.

Copyright (c) 2022 MapLibre contributors

> ®®®

Reviewer: Delete this line?
